modin/core/dataframe/pandas/exchange/dataframe_protocol/column.py

```diff
@@ -0,0 +1,467 @@
+            raise NotImplementedError(f"Columns of type {type(column)} not handled yet")
+
+        # Store the column as a private attribute
+        self._col = column
+        self._allow_copy = allow_copy
+        self._offset = offset
+
+    @property
+    def size(self) -> int:
+        return len(self._col.index)
+
+    @property
+    def offset(self) -> int:
+        return self._offset
+
+    @property
+    def dtype(self) -> Tuple[DTypeKind, int, str, str]:
+        dtype = self._col.dtypes[0]
+
+        if pandas.api.types.is_categorical_dtype(dtype):
+            return (
+                DTypeKind.CATEGORICAL,
+                32,
+                pandas_dtype_to_arrow_c(np.dtype("int32")),
+                "=",
+            )
+        elif pandas.api.types.is_string_dtype(dtype):
+            return (DTypeKind.STRING, 8, pandas_dtype_to_arrow_c(dtype), "=")
+        else:
+            return self._dtype_from_primitive_pandas_dtype(dtype)
+
+    def _dtype_from_primitive_pandas_dtype(
+        self, dtype
+    ) -> Tuple[DTypeKind, int, str, str]:
+        """
+        Deduce dtype specific for the protocol from pandas dtype.
+
+        See `self.dtype` for details.
+
+        Parameters
+        ----------
+        dtype : any
+            A pandas dtype.
+
+        Returns
+        -------
+        tuple
+        """
+        _np_kinds = {
+            "i": DTypeKind.INT,
+            "u": DTypeKind.UINT,
+            "f": DTypeKind.FLOAT,
+            "b": DTypeKind.BOOL,
+        }
+        kind = _np_kinds.get(dtype.kind, None)
+        if kind is None:
+            raise NotImplementedError(
+                f"Data type {dtype} not supported by the dataframe exchange protocol"
+            )
+        return (
+            kind,
+            dtype.itemsize * 8,
+            pandas_dtype_to_arrow_c(dtype),
+            dtype.byteorder,
+        )
+
+    @property
+    def describe_categorical(self) -> Dict[str, Any]:
+        if not self.dtype[0] == DTypeKind.CATEGORICAL:
+            raise RuntimeError(
+                "`describe_categorical only works on a column with "
+                + "categorical dtype!"
+            )
+
+        cat_dtype = self._col.dtypes[0]
+        ordered = cat_dtype.ordered
+        is_dictionary = True
+        # NOTE: this shows the children approach is better, transforming
+        # `categories` to a "mapping" dict is inefficient
+        # codes = self._col.values.codes  # ndarray, length `self.size`
+        # categories.values is ndarray of length n_categories
+        categories = cat_dtype.categories
+        mapping = {ix: val for ix, val in enumerate(categories)}
+        return {
+            "is_ordered": ordered,
+            "is_dictionary": is_dictionary,
+            "mapping": mapping,
+        }
+
+    @property
+    def describe_null(self) -> Tuple[int, Any]:
+        _k = DTypeKind
+        kind = self.dtype[0]
+        value = None
+        if kind == _k.FLOAT:
```

Comment:
when do we need a `_k` instance of the enum, why we can't use it directly?
```suggestion
        kind = self.dtype[0]
        value = None
        if kind ==DTypeKind.FLOAT:
```

Reply:
I agree, the use of DTypeKind directly is reasonable. Changed.